internal/nginx/config/generator.go

```diff
@@ -55,41 +55,41 @@ type executeFunc func(configuration dataplane.Configuration) []byte
 // In case of invalid configuration, NGINX will fail to reload or could be configured with malicious configuration.
 // To validate, use the validators from the validation package.
 func (g GeneratorImpl) Generate(conf dataplane.Configuration) []file.File {
-	files := make([]file.File, 0, len(conf.SSLKeyPairs)+1 /* http configuration file */)
-
-	// Generate PEM files with certificates and keys.
+	files := make([]file.File, 0, len(conf.SSLKeyPairs)+1 /* http config */)
 
 	for id, pair := range conf.SSLKeyPairs {
-		c := make([]byte, 0, len(pair.Cert)+len(pair.Key)+1)
-		c = append(c, pair.Cert...)
-		c = append(c, '\n')
-		c = append(c, pair.Key...)
-
-		files = append(files, file.File{
-			Content: c,
-			Path:    generatePEMFileName(id),
-			Type:    file.TypeSecret,
-		})
+		files = append(files, generatePEM(id, pair.Cert, pair.Key))
 	}
 
-	// Generate HTTP configuration
+	files = append(files, generateHTTPConfig(conf))
+
+	return files
+}
 
-	var generated []byte
+func generatePEM(id dataplane.SSLKeyPairID, cert []byte, key []byte) file.File {
+	c := make([]byte, 0, len(cert)+len(key)+1)
+	c = append(c, cert...)
+	c = append(c, '\n')
+	c = append(c, key...)
+
+	return file.File{
+		Content: c,
+		Path:    filepath.Join(secretsFolder, string(id)+".pem"),
+		Type:    file.TypeSecret,
+	}
+}
+
+func generateHTTPConfig(conf dataplane.Configuration) file.File {
+	var c []byte
 	for _, execute := range getExecuteFuncs() {
-		generated = append(generated, execute(conf)...)
+		c = append(c, execute(conf)...)
 	}
 
-	files = append(files, file.File{
-		Content: generated,
+	return file.File{
+		Content: c,
 		Path:    httpConfigFile,
 		Type:    file.TypeRegular,
-	})
-
-	return files
-}
-
-func generatePEMFileName(id dataplane.SSLKeyPairID) string {
```

Comment:
why get rid of this function?

Reply:
I _feel_ this one is as clear 
```
	return file.File{
		Content: c,
		Path:    filepath.Join(secretsFolder, string(id)+".pem"),
		Type:    file.TypeSecret,
	}
```

as this

```
	return file.File{
		Content: c,
		Path:    generatePEMFileName(id)
		Type:    file.TypeSecret,
	}
```

and we don't need to define that function generatePEMFileName

Reply:
But don't we generate this filename in two places? I just worry that they will get out of sync.

Reply:
oh, I missed that - I inlined using my IDE refactoring tools and also didn't notice it inlined in two files, not one. Thanks for catching!